src/components/Table/TableHead/TableHeader.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { settings } from 'carbon-components';
+import {
+  ArrowUp20 as Arrow,
+  ArrowsVertical20 as Arrows,
+} from '@carbon/icons-react';
+
+import { sortStates } from './state/sorting';
+ 
+const { prefix } = settings;
+
+const translationKeys = {
+  iconDescription: 'carbon.table.header.icon.description',
+};
+
+const translateWithId = (key, { sortDirection, isSortHeader }) => {
+  if (key === translationKeys.iconDescription) {
+    if (isSortHeader) {
+      // When transitioning, we know that the sequence of states is as follows:
+      // NONE -> ASC -> DESC -> NONE
+      if (sortDirection === sortStates.NONE) {
+        return 'Sort rows by this header in ascending order';
+      }
+      if (sortDirection === sortStates.ASC) {
+        return 'Sort rows by this header in descending order';
+      }
+
+      return 'Unsort rows by this header';
+    }
+    return 'Sort rows by this header in ascending order';
+  }
+
+  return '';
+};
+
+const sortDirections = {
+  [sortStates.NONE]: 'none',
+  [sortStates.ASC]: 'ascending',
+  [sortStates.DESC]: 'descending',
+};
+
+const TableHeader = React.forwardRef(function TableHeader(
+  {
+    className: headerClassName,
+    children,
+    isSortable,
+    isSortHeader,
+    onClick,
+    scope,
+    sortDirection,
+    translateWithId: t,
+    ...rest
+  },
+  ref
+) {
+  if (!isSortable) {
+    return (
+      <th {...rest} className={headerClassName} scope={scope}   ref={ref}>
```

Comment:
Added ref property for <th>reference when sorting is disabled.

Reply:
is this a change from Taylor's PR to Carbon? If so, we will want to make a PR for any other changes here so that we do not have to maintain this file.

Reply:
@davidicus this particular change is new, which also need to added in Carbon as earlier change for ref added by @tay1orjones, As  carbon version for taylor is not updated in IOT, so he suggested to create/copy same file in IOT during development (or temporary).

Reply:
Can you raise a PR in Carbon for this  ref={ref} change line 67 and another change at line 88 for {...rest}

Reply:
> @davidicus Can you raise a PR in Carbon for this ref={ref} change line 67 and another change at line 88 for {...rest}.

Please update

Reply:
> > @davidicus Can you raise a PR in Carbon for this ref={ref} change line 67 and another change at line 88 for {...rest}.
> 
> Please update

I am not certain Carbon will allow us to spread props indiscriminately onto the `th`